spec/ruby/optional/capi/set_spec.rb

```diff
@@ -0,0 +1,96 @@
+require_relative 'spec_helper'
+
+load_extension("set")
+
+describe "C-API Set function" do
```

Comment:
Thank you for adding these C API specs, that will definitely help implementing those functions on other Rubies.